codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -266,19 +279,16 @@ public void goAwaySent(final int lastKnownStream, long errorCode, ByteBuf debugD
             }
         }
 
-        try {
-            forEachActiveStream(new Http2StreamVisitor() {
-                @Override
-                public boolean visit(Http2Stream stream) {
-                    if (stream.id() > lastKnownStream && remoteEndpoint.isValidStreamId(stream.id())) {
-                        stream.close();
-                    }
-                    return true;
+        forEachActiveStream(new Http2StreamVisitor() {
+            @Override
+            public boolean visit(Http2Stream stream) {
```

Comment:
Seems like this is the same code as https://github.com/netty/netty/pull/8080/files#diff-9d991ca64301bcd5b3c00d053591f485R244 . Can we just create an `HttpStreamVisitor` imp that we share  by passing the needed stuff via the constructor ?

Reply:
I extracted the iteration into a shared method.